### PR TITLE
Get DevZone username

### PR DESCRIPTION
**Pull request by MagWeaver**
The frontend from nRF Cloud needs to be able to get devzone username

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -322,31 +322,79 @@ namespace Cognito {
...
+            userName = await new Promise<string>((resolve, reject) => {
+                const client = new CognitoSyncManager();
+                client.openOrCreateDataset('identityInfo', (err, dataset) => {
+                    if (err) {
+                        reject(err);
+                    }
+                    dataset
+                        .synchronize({
+                            onSuccess: (dataset) => {
+                                dataset.get('name', (err, value) => {
```
Comment (TheBosZ) [inline src/index.ts]: This should be "email". Look at `login()` about line 226

Comment (MagWeaver) [reply src/index.ts]: I can check to see if that works. I went off of Markus's recommendation.

Comment (MagWeaver) [reply src/index.ts]: The dataset does include email.

**Comment by MagWeaver** [comment]
It would help to know if the user is logged in normally or with DevZone. On the account page has a change password option that couldn't be used if logged in with DevZone. Could this information be in aws-client, in userInfo? For example
```
        return {
            userName,
            accountId,
            loginExpire,
            loginType,
        };
```

**Comment by TheBosZ** [comment]
It could be in there. That's pretty much a free range data store so we can do what we want with it.

**Comment by MagWeaver** [comment]
Ok. I'm going to add a field loginType, that can either have 'cognito' or 'devzone' as its values.

**Comment by TheBosZ** [comment]
I'm going to approve this assuming that it works.

**Pull request merged**